src/LogicApp/LogicApp/Utilities/IntegrationAccountMapOperations.cs

```diff
@@ -99,10 +99,18 @@ public IntegrationAccountMap GetIntegrationAccountMap(string resourceGroupName,
         /// </summary>
         /// <param name="resourceGroupName">The integration account resource group name.</param>
         /// <param name="integrationAccountName">The integration account name.</param>
-        /// <returns>List of integration account schemas.</returns>
-        public IPage<IntegrationAccountMap> ListIntegrationAccountMaps(string resourceGroupName, string integrationAccountName)
+        /// <param name="mapType">The map type to filter by.</param>
+        /// <returns>List of integration account maps.</returns>
+        public IPage<IntegrationAccountMap> ListIntegrationAccountMaps(string resourceGroupName, string integrationAccountName, string mapType)
         {
-            return this.LogicManagementClient.IntegrationAccountMaps.List(resourceGroupName, integrationAccountName, "$filter=mapType eq 'Xslt'&$top=1000");
+            var filter = new Rest.Azure.OData.ODataQuery<IntegrationAccountMapFilter>();
+            if(!string.IsNullOrWhiteSpace(mapType))
+            {
+                filter.Filter = $"MapType eq '{mapType}'";
+            }
+            filter.Top = 1000;
```

Comment:
@refortie how many results were we returning before?

Reply:
We were returning up to 1,000, also 1,000 is the max that the customer can possible have of that resource